test/test_geometry.py

```diff
@@ -441,7 +447,7 @@ def test_populate_bounds_normal(self):
 
         # Convert everything into radians
         vertices_rad = np.radians(vertices_lonlat)
-        vertices_cart = [node_lonlat_rad_to_xyz(v) for v in vertices_rad]
+        vertices_cart = [_lonlat_rad_to_xyz(*v) for v in vertices_rad]
```

Comment:
Is it intentional to keep the for-loop? If not, calling the vectorized function still within a for loop does not look elegant and maybe confusing for the reader. There are many uses similar to this throughout the code.

Reply:
Yes, I intentionally left the code in the test cases iterative for the sake of making the tests compatible with this new implementation.

If you'd like, I can go ahead and try to update all of the existing calls in the test cases, but I didn't see it as a priority. 

I've also left some instances in our code-base untouched for the same reasons, especially in Hongyu's helper functions, since those would require more than a simple plug-in replacement to be vectorized

#730 will aim to replace these in our code base, at least for the helpers related to the face bound computations.

Reply:
Thanks for the clarification! 

#730 sounds good to me regarding the code-base changes. Actually, not only Hongyu's helpers, but I believe we should aim to change all of such calls to vectorized calls for optimization purposes, and I think if it is too much for this PR, it can be handled s part of Rachel's optimization efforts.

The same applies to test case changes; I think we need to address them either as part of this PR or create an issue for that and deal with it soon. I believe the changes would even help with the tests' execution times in our CI action runs, right? I also think the changes should be straightforward to still keep testing for arrays rather than scalars, similar to what you did [here](https://github.com/UXARRAY/uxarray/pull/748/files#r1600845378), right?

Reply:
> https://github.com/UXARRAY/uxarray/issues/730 sounds good to me regarding the code-base changes. Actually, not only Hongyu's helpers, but I believe we should aim to change all of such calls to vectorized calls for optimization purposes, and I think if it is too much for this PR, it can be handled as part of Rachel's optimization efforts.

Agreed, the following modules are the ones I was able to find that still call these functions iteratively. 

* Zonal average helper functions (`integrate.py`)
* Face area helper functions (`area.py`)
* Arc helper functions (`arc.py`)

> The same applies to test case changes; I think we need to address them either as part of this PR or create an issue for that and deal with it soon. BTW, the changes would even help with the tests' execution times in our CI action runs, right? I also think the changes should be straightforward to still keep testing for arrays rather than scalars, similar to what you did [here](https://github.com/UXARRAY/uxarray/pull/748/files#r1600845378), right?

I can make changes to the test cases where appropriate. It should be a simple fix.

Reply:
Sounds good; let me know once you are done, then we can push this forward